src/value/owned.rs

```diff
@@ -610,89 +596,114 @@ impl IonElement for Element {
         }
     }
 
-    fn annotations(&self) -> Self::AnnotationsIterator<'_> {
+    pub fn annotations(&self) -> SymbolsIterator<'_> {
         SymbolsIterator::new(&self.annotations)
     }
 
-    fn with_annotations<I: IntoIterator<Item = Self::SymbolToken>>(self, annotations: I) -> Self {
-        Element::new(annotations.into_iter().collect(), self.value)
+    pub fn with_annotations<S: Into<Symbol>, I: IntoIterator<Item = S>>(
+        self,
+        annotations: I,
+    ) -> Self {
+        let annotations: Vec<Symbol> = annotations.into_iter().map(|i| i.into()).collect();
+        Element::new(annotations, self.value)
     }
 
-    fn has_annotation(&self, annotation: &str) -> bool {
+    pub fn has_annotation(&self, annotation: &str) -> bool {
         self.annotations
             .iter()
             .any(|a| a.text() == Some(annotation))
     }
 
-    fn is_null(&self) -> bool {
+    pub fn is_null(&self) -> bool {
         matches!(&self.value, Value::Null(_))
     }
 
-    fn as_integer(&self) -> Option<&Integer> {
+    pub fn as_integer(&self) -> Option<&Integer> {
         match &self.value {
             Value::Integer(i) => Some(i),
             _ => None,
         }
     }
 
-    fn as_f64(&self) -> Option<f64> {
+    pub fn as_float(&self) -> Option<f64> {
         match &self.value {
             Value::Float(f) => Some(*f),
             _ => None,
         }
     }
 
-    fn as_decimal(&self) -> Option<&Decimal> {
+    pub fn as_decimal(&self) -> Option<&Decimal> {
         match &self.value {
             Value::Decimal(d) => Some(d),
             _ => None,
         }
     }
 
-    fn as_timestamp(&self) -> Option<&Timestamp> {
+    pub fn as_timestamp(&self) -> Option<&Timestamp> {
         match &self.value {
             Value::Timestamp(t) => Some(t),
             _ => None,
         }
     }
 
-    fn as_str(&self) -> Option<&str> {
+    pub fn as_text(&self) -> Option<&str> {
         match &self.value {
             Value::String(text) => Some(text),
             Value::Symbol(sym) => sym.text(),
             _ => None,
         }
     }
 
-    fn as_sym(&self) -> Option<&Self::SymbolToken> {
+    pub fn as_string(&self) -> Option<&str> {
+        match &self.value {
+            Value::String(text) => Some(text),
+            _ => None,
+        }
+    }
+
+    pub fn as_symbol(&self) -> Option<&Symbol> {
         match &self.value {
             Value::Symbol(sym) => Some(sym),
             _ => None,
         }
     }
 
-    fn as_bool(&self) -> Option<bool> {
+    pub fn as_boolean(&self) -> Option<bool> {
         match &self.value {
             Value::Boolean(b) => Some(*b),
             _ => None,
         }
     }
 
-    fn as_bytes(&self) -> Option<&[u8]> {
+    pub fn as_lob(&self) -> Option<&[u8]> {
```

Comment:
🗺️ As with the `string`/`symbol`/`text`, I've broken this into `clob`/`blob`/`lob`.